src/Voron/Data/VoronStream.cs

```diff
@@ -114,10 +131,13 @@ public override int ReadByte()
                     return -1;
             }
 
-            var page = _llt.GetPage(chunk.PageNumber, _pagerRef);
-            _pagerRef.Pager.EnsureMapped(_llt, _pagerRef.PagerPageNumber, VirtualPagerLegacyExtensions.GetNumberOfOverflowPages(page.OverflowSize));
```

Comment:
That was my initial impression, but after talking with @ayende and removing EnsureMapped, it makes sense. Unless of course que can page out a page during the course of it's holding transaction.